src/lib/rules/validate-set-cookie-header/validate-set-cookie-header.ts

```diff
@@ -0,0 +1,317 @@
+
+            return errors;
+        };
+
+        /** Validate `Expire` date format. */
+        const validateExpireDate = (parsedSetCookie: ParsedSetCookieHeader): ValidationMessages => {
+            const cookieName: string = parsedSetCookie.name;
+            const errors: ValidationMessages = [];
+
+            if (!parsedSetCookie.expires) {
+                return errors;
+            }
+
+            // Ref: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Date
+            const utcTimeString: string = new Date(parsedSetCookie.expires).toUTCString();
+            const invalidDateError: string = `Invalid date in 'expires' value of the '${headerName}' header to set '${cookieName}'.`;
+            const invalidDateFormatError: string = `Invalid date format in 'expires' value of the '${headerName}' header to set '${cookieName}'. The recommended format is: ${utcTimeString}`;
+
+            if (utcTimeString === 'Invalid Date') {
+                errors.push(invalidDateError);
+
+                return errors;
+            }
+
+            if (normalizeString(utcTimeString) !== normalizeString(parsedSetCookie.expires)) {
+                errors.push(invalidDateFormatError);
+            }
+
+            return errors;
+        };
+
+        /** Validate the usage of `Max-Age` and `Expires` based on users' browser support matrix */
+        const validateMaxAgeAndExpires = (parsedSetCookie: ParsedSetCookieHeader): ValidationMessages => {
+            const cookieName: string = parsedSetCookie.name;
+            const errors: ValidationMessages = [];
+            const maxAgeCompatibilityMessage: string = `Internet Explorer (IE 6, IE 7, and IE 8) doesn't support 'max-age' directive in the '${headerName}' header to set '${cookieName}'.`;
+            const maxAgeAndExpireDuplicateMessage: string = `The 'max-age' attribute takes precedence when both 'expires' and 'max-age' both exist.`;
+
+            if (supportOlderBrowsers) {
+                // When targetBrowsers contains IE 6, IE 7 or IE 8:
+                // `max-age` can't be used alone.
+                if (parsedSetCookie['max-age'] && !parsedSetCookie.expires) {
+                    errors.push(maxAgeCompatibilityMessage);
+                }
+
+                return errors;
+            }
+
+            // When targetBrowsers only contains modern browsers:
+            // `max-age` takes precedence so `expires` shouldn't be used.
+            // Reference: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#Directives
+            if (parsedSetCookie['max-age'] && parsedSetCookie.expires) {
+                errors.push(maxAgeAndExpireDuplicateMessage);
+            }
+
+            return errors;
+        };
+
+        const loadRuleConfigs = () => {
+            // When `targetedBrowsers` is empty, all browsers are supported.
+            supportOlderBrowsers = context.targetedBrowsers.length === 0 || ['ie 6', 'ie 7', 'ie 8'].some((e) => {
```

Comment:
Let's wait until we decide what to do with https://github.com/sonarwhal/sonar/issues/446#issuecomment-324388133 before merging this PR.